src/session.rs

```diff
@@ -98,19 +128,37 @@ impl Session {
     }
 
     pub fn headers(&self) -> Result<HeaderMap> {
-        Ok(HeaderMap::try_from(&self.content.headers)?)
+        match &self.content.headers {
+            Headers::Map(headers) => Ok(headers.try_into()?),
+            Headers::List(headers) => headers
+                .iter()
+                .map(|Header { name, value }| Ok((name.try_into()?, value.try_into()?)))
+                .collect(),
+        }
     }
 
     pub fn save_headers(&mut self, request_headers: &HeaderMap) -> Result<()> {
+        if let Headers::List(ref mut headers) = self.content.headers {
+            headers.clear();
+        }
+
         for (key, value) in request_headers.iter() {
             let key = key.as_str();
             // HTTPie ignores headers that are specific to a particular request e.g content-length
             // see https://github.com/httpie/httpie/commit/e09b74021c9c955fd7c3bab11f22801aaf9dc1b8
             // we will also ignore cookies as they are taken care of by save_cookies()
             if key != "cookie" && !key.starts_with("content-") && !key.starts_with("if-") {
-                self.content
-                    .headers
-                    .insert(key.into(), value.to_str()?.into());
+                match self.content.headers {
+                    Headers::Map(ref mut headers) => {
+                        headers.insert(key.into(), value.to_str()?.into());
+                    }
```

Comment:
`unreachable!()` sounds fine, as long as we have a test that loads the old format to make sure. This code is a bit misleading.

It would be nice to just not have this case at all but none of the solutions I can think of are a net improvement.

Reply:
Done